google-datacatalog-rdbms-connector/src/google/datacatalog_connectors/rdbms/scrape/metadata_scraper.py

```diff
@@ -77,6 +87,93 @@ def _get_metadata_from_rdbms_connection(self, connection_args, query):
     def _create_dataframe(self, rows):
         return pd.DataFrame(rows)
 
+    def _enrich_metadata_based_on_user_config(self, user_config,
+                                              base_dataframe, connection_args,
+                                              metadata_definition):
+        query_assembler = self._get_query_assembler()
+        enriched_dataframe = base_dataframe
+        if user_config.update_metadata:
+            tbl_names = MetadataNormalizer.get_table_names_from_dataframe(
+                base_dataframe, metadata_definition)
+            update_queries = query_assembler.get_update_queries(tbl_names)
+            logging.info('Updating metadata')
+            self._update_metadata_from_rdbms_connection(
+                connection_args, update_queries)
+        if user_config.scrape_optional_metadata:
+            optional_metadata = user_config.get_chosen_metadata_options()
+            optional_queries = query_assembler.get_optional_queries(
+                optional_metadata)
+            logging.info(
+                'Scraping metadata according to configuration file: {}'.format(
+                    optional_metadata))
+            enriched_dataframe = \
+                self._get_optional_metadata_from_rdbms_connection(
+                    connection_args, optional_queries, base_dataframe,
+                    metadata_definition)
+        return enriched_dataframe
+
+    def _update_metadata_from_rdbms_connection(self, connection_args,
+                                               update_queries):
+        con = None
+        try:
+            con = self._create_rdbms_connection(connection_args)
+            cur = con.cursor()
+            start_update = time.time()
+            for query in update_queries:
+                self._execute_update_query(cur, query)
+            end_update = time.time()
+            logging.info(
+                'Metadata update took {} seconds to run.'
```

Comment:
'recalculating metadata' or 'metadata analysis' perhaps? The point of the message is for the user to understand what's going on.